### PR TITLE
Abstraction improvements

### PR DESCRIPTION
**Pull request by rgnter**
Lobby and ranch directors are now properly isolated, and get data from data director. Data director need futher work, its basically just wrapper around some containers. 
Each director should have handler, or subsystem which can further decouple new mechanics from it. 

Also changed how the address settings work.

### CODE REVIEW

File: 3rd-party/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 add_subdirectory(spdlog)
 add_subdirectory(json)
-add_subdirectory(libpqxx)
+#add_subdirectory(libpqxx)
```
Comment (Miry-0226) [inline 3rd-party/CMakeLists.txt:3]: Why commented? Should it be removed or ?

File: include/server/ranch/DataDirector.hpp
Comment (alborrajo) [inline include/server/ranch/DataDirector.hpp:1]: Is this file duplicated?

File: src/server/DataDirector.cpp
Comment (alborrajo) [inline src/server/DataDirector.cpp:1]: How would this work once we move things to the database

File: 3rd-party/CMakeLists.txt (hunk 1 above)
Comment (rgnter) [reply 3rd-party/CMakeLists.txt:3]: It shouldn't, I just don't have libpqq installed. This will be added back before merge.

**Pull request merged**